### PR TITLE
Added Drafted version of Requirement Update + Status of the Project

### PR DESCRIPTION
**Pull request by Hocng7**

### CODE REVIEW

File: Reporting/M5 Document/Requirement Update.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        -   **Scenario:** Colin, a UBCO student, utilizes Accord to
+            > connect with classmates, collaborate on group projects,
+            > and engage with campus communities. He participates in
+            > discussion servers dedicated to his courses, where he
+            > seeks clarification on assignments, shares study
+            > materials, and discusses course concepts with peers.
+
+    -   **Community Owners/Admin:**
+
+        -   **Scenario:** Immanuel, a community admin for his Helldivers
```
Comment (ColinLefter) [inline Reporting/M5 Document/Requirement Update.md:28]: Excellent scenarios!

File: Reporting/M5 Document/Requirement Update.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    -   Allow the user to add another user as a friend.
+
+    -   Allow users to remove another user from their friend list.
+
+    -   Allow the user to create a group chat.
+
+    -   Allow group chat owners to manage a group chat.
+
+    -   Allow server owners to create a text channel.
```
Comment (ColinLefter) [inline Reporting/M5 Document/Requirement Update.md:71]: This needs to be changed as we don't have the concept of servers anymore. Delete this point.

File: Reporting/M5 Document/Requirement Update.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    -   Allow server owners to edit text channels.
```
Comment (ColinLefter) [inline Reporting/M5 Document/Requirement Update.md:73]: Change to "text channel owners"

File: Reporting/M5 Document/Requirement Update.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    -   Allow users to add other users to the server.
```
Comment (ColinLefter) [inline Reporting/M5 Document/Requirement Update.md:75]: Change to "add other users to a text channel"

File: Reporting/M5 Document/Requirement Update.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    -   Allow server owners to delete a message from a text channel.
```
Comment (ColinLefter) [inline Reporting/M5 Document/Requirement Update.md:77]: Change to "Allow text channel owners to have the ability to delete all messages from a text channel" (this is currently not implemented yet but should be fast)

File: Reporting/M5 Document/Requirement Update.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    -   Allow users to access servers.
```
Comment (ColinLefter) [inline Reporting/M5 Document/Requirement Update.md:79]: Change to "to access text channels"

File: Reporting/M5 Document/Status.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
++---------------------------------------------------------+------------+
+| Allow the user to send text messages.                   | -          |
++---------------------------------------------------------+------------+
+| Allow the user to receive text messages.                | -          |
++---------------------------------------------------------+------------+
+| Allow users to search for other users on the platform.  | -          |
++---------------------------------------------------------+------------+
+| Allow the user to add another user as a friend.         | -          |
++---------------------------------------------------------+------------+
+| Allow users to remove another user from their friend    | -          |
```
Comment (ColinLefter) [inline Reporting/M5 Document/Status.md:31]: Not implemented.

File: Reporting/M5 Document/Status.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+| list.                                                   |            |
++---------------------------------------------------------+------------+
+| Allow the user to create a group chat.                  | -          |
```
Comment (ColinLefter) [inline Reporting/M5 Document/Status.md:34]: Change to "to create a text channel". Users can only create text channels as DMs are created when a friend request is accepted by default. That means the "new text channel" modal only servers to create a channel having at least 3 people in it (DMs are not allowed as they come from your friend list).

**Review by ColinLefter — CHANGES_REQUESTED** [review]
Overall, excellent work! Take a look at my comments and make the necessary modifications. After that, I'll approve it. Awesome job!

File: Reporting/M5 Document/Status.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+ensure the successful implementation of the Accord social platform.
+Regular communication and collaboration between stakeholders, including
+developers, our professor, teaching assistants, and end-users, can help
+refine and iterate on the requirements as the project progresses.
+
+-   [If you have any requirements unimplemented, you will want to
+    > include an extra section showing those requirements have not been
+    > implemented.]{.underline}
+
+    -   So far, the only unimplemented requirement that was documented
```
[the remark this answers is not in the document]
Comment (ColinLefter) [reply Reporting/M5 Document/Status.md:105]: Also, we have 11 tasks in our project backlog and sprint backlog (not including user stories as those are connected to individual tasks).

File: Reporting/M5 Document/Status.md (hunk 7 above)
Comment (Hocng7) [reply Reporting/M5 Document/Status.md:31]: What happened?

File: Reporting/M5 Document/Requirement Update.md (hunk 2 above)
Comment (Hocng7) [reply Reporting/M5 Document/Requirement Update.md:71]: This was taken as the list from actual M2. Not sure if changing it is allowed at this point in the report

File: Reporting/M5 Document/Status.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    > their exact "username".
+
+-   As for our initial user requirements, they provide a comprehensive
+    > overview of the desired functionalities for the Accord social
+    > platform. However, some additional details may be needed to ensure
+    > a thorough understanding of the project scope:
+
+    -   User Account Management:
+
+        -   Password security measures (e.g., password complexity
```
[the remark this answers is not in the document]
Comment (Hocng7) [reply Reporting/M5 Document/Status.md:71]: not quite, I can create account with very non-complicated password such as "abc123456" as long as it reaches the limited number of characters (No requirement for special characters?)

**Comment by ColinLefter** [comment]
Closed since this is already accounted for in our M5 report.

**Pull request closed without merging**